eeprom.c

```diff
@@ -157,25 +157,101 @@ int start_programming(llio_t *self, u32 start_address, int fsize) {
     return 0;
 }
 
-int eeprom_write(llio_t *self, char *bitfile_name, u32 start_address, int fix_boot_flag) {
+int eeprom_write(llio_t *self, char *bitfile_name, u32 start_address, int fix_boot_flag, int sha256_check_flag) {
     board_t *board = self->board;
-    int bytesread, i;
+    int bytesread, i, j;
     u32 eeprom_addr;
     char part_name[32];
     struct stat file_stat;
     FILE *fp;
     struct timeval tv1, tv2;
+    char sha256str[65];
+    char sha256file_name[262];
+    unsigned char sha256in[32];
+    unsigned char sha256bitfile[32];
```

Comment:
Are there named constants that can be used for any of these array bounds?  For instance, `SHA256_DIGEST_LENGTH` in place of 32, and possibly `2*SHA256_DIGEST_LENGTH+1` in place of 65. For `sha256file_name`, maybe `PATH_MAX` though that constant from `<limits.h>` has caveats: https://stackoverflow.com/questions/9449241/where-is-path-max-defined-in-linux#9449307